README.md
# Roboto Serif

Roboto Serif is a serif typeface designed to work alongside Roboto for interfaces and immersive reading, commissioned by Google from Commercial Type. Designed from the ground up as a variable font, Roboto Serif features a full range of weights, widths from condensed to extended, and optical sizes from micro to large display. It also includes a grade axis to fine tune contrast for different usages without affecting line breaks and copyfit.

This family supports the Google Fonts Latin Pro glyph set, for use in over 130 languages including Vietnamese.


## Install dependencies

```
python3 -m venv venv
source venv/bin/activate
pip install -r requirements.txt
```

## Building fonts

```
cd sources
sh build.sh
```


## Historical repo

The origin repo can be found [here](https://github.com/CommercialType/RobotoSerif) which contains development directories chronicling the typeface's development.


CHANGELOG.txt
FONTLOG for Roboto Serif Text

ChangeLog


Roboto Serif v1.004 (12 April 2021)
- Release to public


Roboto Serif v0.004 (11 September 2020)

- Sources: Added 24 sources to improve grad axis across all optical sizes
- License: Updated license in fonts with text from OFL.txt
- Sources: combining accents set to zero width, negative offset.


Roboto Serif v0.003 (17 June 2020)

- Family Name: rename from "Roboto Serif Text" to "Roboto Serif"
- glyph outlines: Moved overlapped points in the apex of A (and other characters with similar problem) closer together so they wouldn’t poke out in the grade deltas.
- width: respec range: 50 : 150, default 100
- weight: respec range: 100 : 900, default 400
- weight: instances at 
	100 Thin
	200 Extralight
	300 Light
	400 Regular
	500 Medium
	600 SemiBold
	700 Bold
	* NOTE: no ExtraBold
	900 Black
- weight: fix typo: "wgth" renamed to "wght"
- optical size: respec range: 8px (Micro), 20px (Text), 144px (Big)
- optical size: instances at 8px, 14px, 20px, 60px, 100px
- optical size: label renamed to "Optical Size"
- grade: respec range: -1 : 1
- instances: reordered by: opsz/wght/wdth/GRAD
- designspace: added "name" field (to help if anyone wants to build specific instances) 
- designspace: added "stylemapfamilyname" and "stylemapstylename" to merge Bolds with Regulars
- build_static.sh: added
- build_variable.sh: updated to be more robust


Roboto Serif Text v0.002 (4 May 2020)
- copyright change to "Copyright 2020 The Roboto Serif Project Authors" to match Google spec.
- rename instance names to match Google spec.
- fix glyph names: thai:baht renamed to baht, thai:baht.proportional renamed to baht.proportional
- change file names to to RobotoSerifText[grad,opsz,wdth,wgth].ttf and RobotoSerifText-Italic[grad,opsz,wdth,wgth].ttf to match Google spec.
- update metrics to match Roboto Release v2.138
openTypeHheaAscender = 928
openTypeHheaDescender = -244
openTypeHheaLineGap = 0
openTypeOS2TypoAscender = 1048
openTypeOS2TypoDescender = -271
openTypeOS2TypoLineGap = 0
openTypeOS2WinAscent = 1048
openTypeOS2WinDescent = 251
- update underline thickness and position to match Roboto Release v2.138
postscriptUnderlinePosition = -150
postscriptUnderlineThickness = 100
- change license from Apache to OFL.
- version minor: 1.002
- add commands to build.sh: gftools fix-nonhinting and gftools fix-dsig


Roboto Serif Text v0.001 (20 April 2020)
- Initial release


sources/fix.py
from glob import glob
from ufoLib2 import Font

sources = glob("*.ufo")

for source in sources:
    source = Font(source)

    if not source.info.openTypeOS2Selection:
        source.info.openTypeOS2Selection = []
    if 7 not in source.info.openTypeOS2Selection:
        source.info.openTypeOS2Selection.append(7)

    if "openTypeOS2CodePageRanges":
        delattr(source.info, "openTypeOS2CodePageRanges")

    if "openTypeNameLicense":
        delattr(source.info, "openTypeNameLicense")

    if "openTypeOS2UnicodeRanges":
        delattr(source.info, "openTypeOS2UnicodeRanges")

    source.info.openTypeOS2TypoAscender = 750
    source.info.openTypeOS2TypoDescender = -250
    source.info.openTypeOS2TypoLineGap = 50

    source.info.openTypeHheaAscender = 750
    source.info.openTypeHheaDescender = -250
    source.info.openTypeHheaLineGap = 50
    source.save()


sources/fix_names.py
# TODO Drop this and update designspace + ufos sources properly
from fontTools.ttLib import TTFont
import os

fonts_dir = os.path.join(os.path.dirname(__file__), "..", "fonts", "variable")
# TODO get from ufos
version = 1.004
license = "This Font Software is licensed under the SIL Open Font License, Version 1.1. This license is available with a FAQ at: https://scripts.sil.org/OFL"

# Roman
roman = TTFont(os.path.join(fonts_dir, "RobotoSerif[GRAD,opsz,wdth,wght].ttf"))
roman_name = roman['name']
roman_name.setName("Roboto Serif 20pt", 1, 3, 1, 0x409)
roman_name.setName("Regular", 2, 3, 1, 0x409)
roman_name.setName(f"{version};COMM;RobotoSerif-20ptRegular", 3, 3, 1, 0x409)
roman_name.setName("Roboto Serif 20pt Regular", 4, 3, 1, 0x409)
roman_name.setName("RobotoSerif-20ptRegular", 6, 3, 1, 0x409)
roman_name.setName(license, 13, 3, 1, 0x409)
roman_name.setName("Roboto Serif", 16, 3, 1, 0x409)
roman_name.setName("20pt Regular", 17, 3, 1, 0x409)

# Italic
italic = TTFont(os.path.join(fonts_dir, "RobotoSerif-Italic[GRAD,opsz,wdth,wght].ttf" ))
italic_name = italic['name']
italic_name.setName("Roboto Serif 20pt", 1, 3, 1, 0x409)
italic_name.setName("Italic", 2, 3, 1, 0x409)
italic_name.setName(f"{version};COMM;RobotoSerifNormalItalic-Regular", 3, 3, 1, 0x409)
italic_name.setName("Roboto Serif 20pt Italic", 4, 3, 1, 0x409)
italic_name.setName("RobotoSerif-20ptItalic", 6, 3, 1, 0x409)
italic_name.setName(license, 13, 3, 1, 0x409)
italic_name.setName("Roboto Serif", 16, 3, 1, 0x409)
italic_name.setName("20pt Italic", 17, 3, 1, 0x409)


roman.save(roman.reader.file.name)
italic.save(italic.reader.file.name)
print("Done fixing names")


sources/designspace_cleanup.py
"""
Create a GF Compliant designspace
"""
import sys
from fontTools.designspaceLib import DesignSpaceDocument, InstanceDescriptor

WEIGHT = {
    100: "Thin",
    200: "ExtraLight",
    300: "Light",
    400: "Regular",
    500: "Medium",
    600: "SemiBold",
    700: "Bold",
    800: "ExtraBold",
    900: "Black",
}


if len(sys.argv) != 2:
    print("Usage: python designspace_cleanup.py mydesignspace.designspace")
    sys.exit(1)

designspace_file = sys.argv[1]

ds = DesignSpaceDocument()
ds.read(designspace_file)

is_italic = "Italic" in designspace_file or any(a.tag in ["ital", "slnt"] for a in ds.axes)
dflt_coords = {a.name: a.default for a in ds.axes}
wght_axis = next((a for a in ds.axes if a.tag == "wght"), None)
wght_min = wght_axis.minimum
wght_max = wght_axis.maximum


new_instances = []
for wght in range(int(wght_min), int(wght_max)+100, 100):
    i = InstanceDescriptor()
    i.familyName = "Roboto Serif"
    location = dflt_coords
    location["weight"] = wght

    if is_italic:
        i.styleName = f"{WEIGHT[wght]} Italic"
    else:
        i.styleName = f"{WEIGHT[wght]}"
    i.location = location
    new_instances.append(i)


ds.instances = new_instances
ds.write(designspace_file)
print("Updated designspace")
